### DISCUSSION TITLE
Using container builds with private feeds

### DISCUSSION THREAD

**Discussion by jamiewinder** [aws/aws-extensions-for-dotnet-cli#300] [Q&A]
When using the Lambda tools such as `dotnet lambda package` which results in a container build (e.g. AOT), what is the recommended approach when your project referencing privates Nuget feeds?

The [https://github.com/dotnet/dotnet-docker/blob/main/documentation/scenarios/nuget-credentials.md#using-the-azure-artifact-credential-provider](.NET docs) that relate to this lean heavily towards using a credentials provider which - since I'm using Azure DevOps for both the pipeline and the feed - makes a lot of sense. However it isn't obvious to me how / whether it's possible as all to do this with the CLI extensions.

**Comment by ashishdhingra** [comment]
@jamiewinder I also came across [Managing NuGet Credentials in Docker Scenarios](https://github.com/dotnet/dotnet-docker/blob/main/documentation/scenarios/nuget-credentials.md), which appears to be one of the approach to follow. Below are some references on AWS side;
- [Using NuGet with AWS CodeArtifact](https://aws.amazon.com/blogs/devops/using-nuget-with-aws-codeartifact/)
- [Publish NuGet packages via Docker Release Container](https://nikiforovall.github.io/dotnet/nuget/docker/2021/05/09/nuget-release-container.html)

Hope these help.

Thanks,
Ashish

**Reply by jamiewinder** [reply]
Thanks @ashishdhingra .

I have had some luck using `NuGetPackageSourceCredentials_X` environment variables (where X is a feed name) to provide the credentials, but only when calling `dotnet publish` myself. I'm not sure I can pass these environment variables to the `dotnet lambda package` command for it to forward to dotnet publish?

**Comment by sasha-simplyture** [comment]
Hi everyone.
Any news on this topic? In my opinion, this is **quite critical**, as organizations **often use private Nuget feeds** (repositories).

For now, I see only **one way to do this**: use the docker image (public.ecr.aws/sam/build-dotnet8 or lambda) without `aws-extensions-for-dotnet-cli` at all.
